diff --git a/dist/index.d.ts b/dist/index.d.ts
index 664d9ccc7b6df08583bbcf32774bc24c6f9e5b02..eaef9636178e037d5dfc8dad931703ef21ecd905 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,5 +1,5 @@
-import * as playwright from 'playwright';
-import { Browser, BrowserContextOptions, LaunchOptions } from 'playwright';
+import * as playwright from 'playwright-core';
+import { Browser, BrowserContextOptions, LaunchOptions } from 'playwright-core';
 import { NuxtConfig, Nuxt } from '@nuxt/schema';
 import { ExecaChildProcess } from 'execa';
 import { FetchOptions } from 'ofetch';
diff --git a/dist/index.mjs b/dist/index.mjs
index 094cc692581c8e8faf46ee178c6dab33a98aaef0..215b31bbdde2f19572f233e1439c30a05d1108b7 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -15,11 +15,11 @@ async function createBrowser() {
   const ctx = useTestContext();
   let playwright;
   try {
-    playwright = await import(String("playwright"));
+    playwright = await import(String("playwright-core"));
   } catch {
     throw new Error(`
-      The dependency 'playwright' not found.
-      Please run 'yarn add --dev playwright' or 'npm install --save-dev playwright'
+      The dependency 'playwright-core' not found.
+      Please run 'yarn add --dev playwright-core' or 'npm install --save-dev playwright-core'
     `);
   }
   const { type, launch } = ctx.options.browserOptions;
diff --git a/package.json b/package.json
index b8102a407c65306f7d1909e1231d3ee035d5c652..28ec668e8e3207bd19b67f7c65e87090defdcaef 100644
--- a/package.json
+++ b/package.json
@@ -31,13 +31,13 @@
   },
   "devDependencies": {
     "@jest/globals": "29.6.0",
-    "playwright": "1.35.1",
+    "playwright-core": "1.35.1",
     "unbuild": "latest",
     "vitest": "0.32.4"
   },
   "peerDependencies": {
     "@jest/globals": "^29.5.0",
-    "playwright": "^1.34.3",
+    "playwright-core": "^1.34.3",
     "vitest": "^0.30.0 || ^0.31.0 || ^0.32.0",
     "vue": "^3.3.4"
   },
@@ -45,7 +45,7 @@
     "@jest/globals": {
       "optional": true
     },
-    "playwright": {
+    "playwright-core": {
       "optional": true
     },
     "vitest": {