diff --git a/dist/runtime/internal/config.mjs b/dist/runtime/internal/config.mjs
index 8a81bda77e1009c7e9c152c6205894da68851031..9afeee0deff76ea92d8fabb998121074d3d0c720 100644
--- a/dist/runtime/internal/config.mjs
+++ b/dist/runtime/internal/config.mjs
@@ -2,6 +2,7 @@ import { klona } from "klona";
 import { appConfig as _inlineAppConfig } from "#nitro-internal-virtual/app-config";
 import { applyEnv } from "./utils.env.mjs";
 const _inlineRuntimeConfig = process.env.RUNTIME_CONFIG;
+console.log('inlined runtime config', _inlineRuntimeConfig)
 const envOptions = {
   prefix: "NITRO_",
   altPrefix: _inlineRuntimeConfig.nitro.envPrefix ?? process.env.NITRO_ENV_PREFIX ?? "_",
@@ -20,6 +21,7 @@ export function useRuntimeConfig(event) {
   const runtimeConfig = klona(_inlineRuntimeConfig);
   applyEnv(runtimeConfig, envOptions);
   event.context.nitro.runtimeConfig = runtimeConfig;
+  console.log('resolved runtime config', runtimeConfig)
   return runtimeConfig;
 }
 const _sharedAppConfig = _deepFreeze(klona(_inlineAppConfig));
