diff --git a/lib/esm/nuxt-plugin.js b/lib/esm/nuxt-plugin.js
index 0a71f65b04fb4cb26cbdd5ebdca92962e62f55d5..630b323bf505bf9f9c72f9108d4ff5f666cb3cd3 100644
--- a/lib/esm/nuxt-plugin.js
+++ b/lib/esm/nuxt-plugin.js
@@ -16,7 +16,11 @@ const PlausiblePlugin = (context, inject) => {
         if (options.enableAutoOutboundTracking === true) {
             plausible.enableAutoOutboundTracking();
         }
-        inject('plausible', plausible);
+        if (inject) {
+            inject('plausible', plausible);
+        } else {
+            context.provide('plausible', plausible)
+        }
     }
 };
 export default PlausiblePlugin;